server/tracker.go

```diff
@@ -110,7 +111,10 @@ func (t *Tracker) remove(conn net.Conn) {
 
 // banned returns true if the player has been banned.
 func (t *Tracker) banned(data *trackerData) bool {
-	return t.poolSize == (len(data.bannedBy) + 1)
+	data.mutex.Lock()
+	defer data.mutex.Unlock()
```

Comment:
Just to make sure I understand correctly, all of the tests like this one *could* be done with RLock() right?

Reply:
Yep, many of these could probably use RLock(). Something we should probably clean up in a future PR.

Reply:
I went ahead and fixed all the locks. =)